internal/models/user.go

```diff
@@ -239,13 +239,24 @@ func (u *User) SetPhone(tx *storage.Connection, phone string) error {
 	return tx.UpdateOnly(u, "phone")
 }
 
-// UpdatePassword updates the user's password
-func (u *User) UpdatePassword(tx *storage.Connection, password string, sessionID *uuid.UUID) error {
-	pw, err := crypto.GenerateFromPassword(context.Background(), password)
+func (u *User) SetPassword(ctx context.Context, password string) error {
+	if password == "" {
+		u.EncryptedPassword = ""
+		return nil
+	}
+
+	pw, err := crypto.GenerateFromPassword(ctx, password)
 	if err != nil {
 		return err
 	}
+
 	u.EncryptedPassword = pw
+
+	return nil
+}
+
+// UpdatePassword updates the user's password. Use SetPassword outside of a transaction first!
```

Comment:
can we make it more explicit in the comment that it is intended that `UpdatePassword` doesn't hash the password?

Reply:
also, this seems rather dangerous if someone in the future forgets to call `SetPassword` before `UpdatePassword` 

is there a way to do some validation inside `UpdatePassword` to check if the password is hashed?

Reply:
I don't think there's a way. We kinda just have to make sure we always check that behavior... Ideally we can refactor this to not use `UpdatePassword` as it just does `UpdateOnly` in the background... or refactor it in a different way.